### PR TITLE
Fixing notice when field not in submitted data

### PR DESCRIPTION
**Pull request by aniracode**
A notice appears when the entity is submitted without the field

### CODE REVIEW

File: src/Model/Behavior/UploadBehavior.php [hunk 1]
```diff
@@ -75,7 +75,10 @@ public function beforeMarshal(EventInterface $event, ArrayObject $data, ArrayObj
             if (!empty($dataArray[$field]) && $dataArray[$field]->getError() !== UPLOAD_ERR_NO_FILE) {
                 continue;
             }
-            unset($data[$field]);
+
+            if(isset($data[$field])) {
```
Comment (ADmad) [inline src/Model/Behavior/UploadBehavior.php:79]: Space needed after `if`.

**Comment by ADmad** [comment]
Silly PHP doesn't throw notices in `unset()` for non-existent keys of arrays, but does so for `ArrayAccess`/`ArrayObject`.

**Comment by ADmad** [comment]
A test case would be nice.

**Comment by aniracode** [comment]
I'm a newbie with tests. I'll give it a try...

**Comment by ADmad** [comment]
Fixed in #542

**Pull request closed without merging**